runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -1052,14 +1051,12 @@ protected final void invokeFinishBundle() {
         LOG.debug("Invoking bundle finish callback.");
         bundleFinishedCallback.run();
       }
-    }
-    try {
-      if (previousBundleFinishTime - getProcessingTimeService().getCurrentProcessingTime()
-          > maxBundleTimeMills) {
```

Comment:
Would it be possible to add test ensuring that watermark is always processed on finish bundle?